File: Haskell/beginner/1180.hs
-- https://judge.beecrowd.com/pt/problems/view/1180
-- Menor e Posição

-- Imports
import Control.Monad (replicateM)
import Data.List (minimumBy)
import Data.Ord (comparing)

main :: IO ()
main = do
    --Read array size
    arraySize <- readLn :: IO Int

    -- Reads n array values
    -- Reading Values in a line
    valuesLine <- getLine
    let values = map read (words valuesLine) :: [Int]

    -- Finds the minimum value and its position
    let indexed = zip [0..] values  -- [(posição, valor)]
        (pos, minVal) = minimumBy (comparing snd) indexed

    -- Print results
    putStrLn $ "Menor valor: " ++ show minVal
    putStrLn $ "Posicao: " ++ show pos
    
-- By: Fernando Serra
-- https://github.com/fernandosserra

File: Python/beginner/1180.py
# https://judge.beecrowd.com/pt/problems/view/1180
# Menor e Posição

# Readin a integer to set the array size
array_size = int(input())

# Reading n integers from the user
values = list(map(int, input().split()))

# Finding the minimum value and its position
min_value = min(values)
position = values.index(min_value)

# Printing the results
print(f"Menor valor: {min_value}")
print(f"Posicao: {position}")

# By: Fernando Serra
# https://github.com/fernandosserra